### ISSUE TITLE
SC2031 false positive

### ISSUE THREAD

**Issue by thockin** [koalaman/shellcheck#2877]
This is a weird one.  When I run shellcheck on a reduced form of this case, it does not trigger.  When I run it in situ, it does.

#### For bugs
- Rule Id:  SC2031
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

  # (in bash, in a function)
  # Try to normalize input names.
  local -a targets
  while IFS="" read -r target; do
    targets+=("$target")                                                                                                      
  done < <(kube::golang::normalize_go_targets "$@")
```

#### Here's what shellcheck currently says:

$ docker run --rm -v /home/thockin/src/kubernetes:/home/thockin/src/kubernetes -w /home/thockin/src/kubernetes docker.io/koalaman/shellcheck-alpine:v0.9.0 shellcheck --external-sources --exclude=1090,1091,2230 --color=auto hack/make-rules/test.sh

In hack/make-rules/test.sh line 223:
    targets+=("$target")
    ^-----^ SC2031 (info): targets was modified in a subshell. That change might be lost.

For more information:
  https://www.shellcheck.net/wiki/SC2031 -- targets was modified in a subshel...

#### Here's what I wanted or expected to see:

Nothing

**Comment by thockin** [comment]
Actually, it seems to depend on how shellcheck is called:

This triggers (falsely):
```
$ docker run --rm -v /home/thockin/src/kubernetes:/home/thockin/src/kubernetes -w /home/thockin/src/kubernetes docker.io/koalaman/shellcheck-alpine:v0.9.0 shellcheck --external-sources --exclude=1090,1091,2230 hack/make-rules/test.sh

In hack/make-rules/test.sh line 224:
    targets+=("$target")
    ^-----^ SC2031 (info): targets was modified in a subshell. That change might be lost.

For more information:
  https://www.shellcheck.net/wiki/SC2031 -- targets was modified in a subshel...
```

But this does not:
```
$ docker run --rm -v /home/thockin/src/kubernetes:/home/thockin/src/kubernetes -w /home/thockin/src/kubernetes/hack/make-rules docker.io/koalaman/shellcheck-alpine:v0.9.0 shellcheck --external-sources --exclude=1090,1091,2230 ./test.sh
```

The only difference is the working dir and the path to the file
